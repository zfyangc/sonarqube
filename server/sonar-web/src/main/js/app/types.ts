/*
 * SonarQube
 * Copyright (C) 2009-2016 SonarSource SA
 * mailto:contact AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
export enum BranchType {
  LONG = 'LONG',
  SHORT = 'SHORT'
}

export interface MainBranch {
  analysisDate?: string;
  isMain: true;
  name: string;
  status?: {
    qualityGateStatus: string;
  };
}

export interface LongLivingBranch {
  analysisDate?: string;
  isMain: false;
  name: string;
  status?: {
    qualityGateStatus: string;
  };
  type: BranchType.LONG;
}

export interface ShortLivingBranch {
  analysisDate?: string;
  isMain: false;
  isOrphan?: true;
  mergeBranch: string;
  name: string;
  status?: {
    bugs: number;
    codeSmells: number;
    vulnerabilities: number;
  };
  type: BranchType.SHORT;
}

export type Branch = MainBranch | LongLivingBranch | ShortLivingBranch;

export interface Extension {
  key: string;
  name: string;
}

export interface Component {
  analysisDate?: string;
  breadcrumbs: Array<{
    key: string;
    name: string;
    qualifier: string;
  }>;
  configuration?: ComponentConfiguration;
  description?: string;
  extensions?: Extension[];
  isFavorite?: boolean;
  key: string;
  name: string;
  organization: string;
  path?: string;
  qualifier: string;
  refKey?: string;
  version?: string;
}

interface ComponentConfiguration {
  extensions?: Extension[];
  showBackgroundTasks?: boolean;
  showLinks?: boolean;
  showManualMeasures?: boolean;
  showQualityGates?: boolean;
  showQualityProfiles?: boolean;
  showPermissions?: boolean;
  showSettings?: boolean;
  showUpdateKey?: boolean;
}

export interface Metric {
  custom?: boolean;
  decimalScale?: number;
  description?: string;
  direction?: number;
  domain?: string;
  hidden?: boolean;
  key: string;
  name: string;
  qualitative?: boolean;
  type: string;
}

export interface Organization {
  adminPages?: { key: string; name: string }[];
  avatar?: string;
  canAdmin?: boolean;
  canDelete?: boolean;
  canProvisionProjects?: boolean;
  canUpdateProjectsVisibilityToPrivate?: boolean;
  description?: string;
  isDefault?: boolean;
  key: string;
  name: string;
  pages?: { key: string; name: string }[];
  projectVisibility: Visibility;
  url?: string;
}

export interface Paging {
  pageIndex: number;
  pageSize: number;
  total: number;
}

export enum Visibility {
  Public = 'public',
  Private = 'private'
}

export interface CurrentUser {
  isLoggedIn: boolean;
  showOnboardingTutorial?: boolean;
}
