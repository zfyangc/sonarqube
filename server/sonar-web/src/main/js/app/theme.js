/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

// IMPORTANT: any change in this file requires restart of the dev server

module.exports = {
  // colors
  blue: '#4b9fd5',
  lightBlue: '#cae3f2',
  darkBlue: '#236a97',
  green: '#00aa00',
  lightGreen: '#b0d513',
  yellow: '#eabe06',
  orange: '#ed7d20',
  red: '#d4333f',
  purple: '#9139d4',

  gray94: '#efefef',
  gray80: '#cdcdcd',
  gray71: '#b4b4b4',
  gray67: '#aaa',
  gray60: '#999',

  barBackgroundColor: '#f3f3f3',
  barBorderColor: '#e6e6e6',

  baseFontColor: '#444',
  secondFontColor: '#777',

  leakColor: '#fbf3d5',
  leakBorderColor: '#eae3c7',

  // sizes
  baseFontSize: '13px',
  smallFontSize: '12px',
  mediumFontSize: '14px',
  bigFontSize: '16px',

  controlHeight: '24px',
  smallControlHeight: '20px',
  tinyControlHeight: '16px',

  globalNavHeight: '48px',
  globalNavHeightRaw: 48,
  globalNavContentHeight: '32px',
  globalNavContentHeightRaw: 32,

  // different
  defaultShadow: '0 6px 12px rgba(0, 0, 0, 0.175)',

  // z-index
  // =======
  //    1 -  100  for page elements (e.g. sidebars, panels)
  //  101 -  500  for generic page fixed elements (e.g. navigation, workspace)
  //  501 - 3000  for page ui elements
  // 3001 - 8000  for generic ui elements (e.g. dropdowns, tooltips)

  // common
  aboveNormalZIndex: '3',
  normalZIndex: '2',
  belowNormalZIndex: '1',

  // ui elements
  tooltipZIndex: '8000',

  dropdownMenuZIndex: '7500',

  processContainerZIndex: '7000',

  modalZIndex: '6001',
  modalOverlayZIndex: '6000',

  bubblePopupZIndex: '5000'
};
