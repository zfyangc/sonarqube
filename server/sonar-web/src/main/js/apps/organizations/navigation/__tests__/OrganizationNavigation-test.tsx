/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import OrganizationNavigation from '../OrganizationNavigation';
import { Visibility } from '../../../../app/types';

jest.mock('../../../issues/utils', () => ({
  isMySet: () => false
}));

const organization = {
  key: 'foo',
  name: 'Foo',
  canAdmin: false,
  canDelete: false,
  projectVisibility: Visibility.Public
};

it('regular user', () => {
  expect(getWrapper()).toMatchSnapshot();
});

it('admin', () => {
  expect(
    getWrapper({ organization: { ...organization, canAdmin: true, canDelete: true } })
  ).toMatchSnapshot();
});

it('undeletable org', () => {
  expect(
    getWrapper({ organization: { ...organization, canAdmin: true, canDelete: false } })
  ).toMatchSnapshot();
});

function getWrapper(props = {}) {
  return shallow(
    <OrganizationNavigation
      location={{ pathname: '/organizations/foo' }}
      organization={organization}
      {...props}
    />
  );
}
