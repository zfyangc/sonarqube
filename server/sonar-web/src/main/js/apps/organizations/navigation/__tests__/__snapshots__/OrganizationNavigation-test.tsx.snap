// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin 1`] = `
<ContextNavBar
  height={65}
  id="context-navigation"
>
  <div
    className="navbar-context-header"
  >
    <h1
      className="display-inline-block"
    >
      <OrganizationIcon
        className="little-spacer-right"
      />
      <Link
        className="link-base-color link-no-underline"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo"
      >
        <strong>
          Foo
        </strong>
      </Link>
    </h1>
  </div>
  <div
    className="navbar-context-meta"
  />
  <NavBarTabs>
    <li>
      <Link
        className=""
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/projects"
      >
        projects.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to={
          Object {
            "pathname": "/organizations/foo/issues",
            "query": Object {
              "resolved": "false",
            },
          }
        }
      >
        issues.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/quality_profiles"
      >
        quality_profiles.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/rules"
      >
        coding_rules.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to={
          Object {
            "pathname": "/organizations/foo/quality_gates",
          }
        }
      >
        quality_gates.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/members"
      >
        organization.members.page
      </Link>
    </li>
    <li
      className="dropdown"
    >
      <a
        className="dropdown-toggle"
        data-toggle="dropdown"
        href="#"
        id="organization-navigation-admin"
      >
        layout.settings
         
        <i
          className="icon-dropdown"
        />
      </a>
      <ul
        className="dropdown-menu"
      >
        <li>
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/organizations/foo/groups"
          >
            user_groups.page
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/organizations/foo/permissions"
          >
            permissions.page
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/organizations/foo/permission_templates"
          >
            permission_templates
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/organizations/foo/projects_management"
          >
            projects_management
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/organizations/foo/edit"
          >
            edit
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/organizations/foo/delete"
          >
            delete
          </Link>
        </li>
      </ul>
    </li>
  </NavBarTabs>
</ContextNavBar>
`;

exports[`regular user 1`] = `
<ContextNavBar
  height={65}
  id="context-navigation"
>
  <div
    className="navbar-context-header"
  >
    <h1
      className="display-inline-block"
    >
      <OrganizationIcon
        className="little-spacer-right"
      />
      <Link
        className="link-base-color link-no-underline"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo"
      >
        <strong>
          Foo
        </strong>
      </Link>
    </h1>
  </div>
  <div
    className="navbar-context-meta"
  />
  <NavBarTabs>
    <li>
      <Link
        className=""
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/projects"
      >
        projects.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to={
          Object {
            "pathname": "/organizations/foo/issues",
            "query": Object {
              "resolved": "false",
            },
          }
        }
      >
        issues.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/quality_profiles"
      >
        quality_profiles.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/rules"
      >
        coding_rules.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to={
          Object {
            "pathname": "/organizations/foo/quality_gates",
          }
        }
      >
        quality_gates.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/members"
      >
        organization.members.page
      </Link>
    </li>
  </NavBarTabs>
</ContextNavBar>
`;

exports[`undeletable org 1`] = `
<ContextNavBar
  height={65}
  id="context-navigation"
>
  <div
    className="navbar-context-header"
  >
    <h1
      className="display-inline-block"
    >
      <OrganizationIcon
        className="little-spacer-right"
      />
      <Link
        className="link-base-color link-no-underline"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo"
      >
        <strong>
          Foo
        </strong>
      </Link>
    </h1>
  </div>
  <div
    className="navbar-context-meta"
  />
  <NavBarTabs>
    <li>
      <Link
        className=""
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/projects"
      >
        projects.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to={
          Object {
            "pathname": "/organizations/foo/issues",
            "query": Object {
              "resolved": "false",
            },
          }
        }
      >
        issues.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/quality_profiles"
      >
        quality_profiles.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/rules"
      >
        coding_rules.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to={
          Object {
            "pathname": "/organizations/foo/quality_gates",
          }
        }
      >
        quality_gates.page
      </Link>
    </li>
    <li>
      <Link
        activeClassName="active"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/organizations/foo/members"
      >
        organization.members.page
      </Link>
    </li>
    <li
      className="dropdown"
    >
      <a
        className="dropdown-toggle"
        data-toggle="dropdown"
        href="#"
        id="organization-navigation-admin"
      >
        layout.settings
         
        <i
          className="icon-dropdown"
        />
      </a>
      <ul
        className="dropdown-menu"
      >
        <li>
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/organizations/foo/groups"
          >
            user_groups.page
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/organizations/foo/permissions"
          >
            permissions.page
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/organizations/foo/permission_templates"
          >
            permission_templates
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/organizations/foo/projects_management"
          >
            projects_management
          </Link>
        </li>
        <li>
          <Link
            activeClassName="active"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/organizations/foo/edit"
          >
            edit
          </Link>
        </li>
      </ul>
    </li>
  </NavBarTabs>
</ContextNavBar>
`;
